File: Sis104Avance/app/src/main/java/com/example/sis104avance/ejercicios/Ecuacion.java
package com.example.sis104avance.ejercicios;

public class Ecuacion {

    private double a, b, c;

    public Ecuacion(double a, double b, double c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    // Método para calcular el discriminante
    private Double D() {
        return (this.b * this.b) - (4 * this.a * this.c);
    }

    // Método para calcular las raíces
    public String Raices() {
        String solucion;
        Double discriminante = D();

        if (discriminante == 0) {
            Double x = -this.b / (2 * this.a);  // Se corrige la fórmula
            solucion = "X1 = X2 = " + x;
        } else if (discriminante > 0) {
            Double x1 = (-this.b + Math.sqrt(discriminante)) / (2 * this.a);  // Se corrigió el paréntesis y la fórmula
            Double x2 = (-this.b - Math.sqrt(discriminante)) / (2 * this.a);
            solucion = "X1 = " + x1 + " , X2 = " + x2;
        } else {
            Double parteReal = -this.b / (2 * this.a);
            Double parteImag = Math.sqrt(Math.abs(discriminante)) / (2 * this.a);
            solucion = "X1 = " + parteReal + " + " + parteImag + "i, X2 = " + parteReal + " - " + parteImag + "i";
        }

        return solucion;
    }
}

File: trabajoSIS104/app/src/main/java/com/example/trabajosis104/ejercicios/Ecuacion.kt
package com.example.trabajosis104.ejercicios

import kotlin.math.sqrt

class Ecuacion(private val a: Double, private val b: Double, private val c: Double) {

    // Método para calcular el discriminante
    private fun discriminante(): Double {
        return (b * b) - (4 * a * c)
    }

    // Método para calcular las raíces
    fun calcularRaices(): String {
        val discriminante = discriminante()
        val solucion: String

        // Si el discriminante es cero, una única raíz real
        if (discriminante == 0.0) {
            val x = -b / (2 * a)
            solucion = "X1 = X2 = $x"
        }
        // Si el discriminante es mayor que cero, dos raíces reales distintas
        else if (discriminante > 0) {
            val x1 = (-b + sqrt(discriminante)) / (2 * a)
            val x2 = (-b - sqrt(discriminante)) / (2 * a)
            solucion = "X1 = $x1 , X2 = $x2"
        }
        // Si el discriminante es negativo, soluciones complejas
        else {
            val parteReal = -b / (2 * a)
            val parteImag = sqrt(-discriminante) / (2 * a)
            solucion = "X1 = $parteReal + ${parteImag}i, X2 = $parteReal - ${parteImag}i"
        }

        return solucion
    }
}
